R/geom-segment.r

```diff
@@ -42,6 +43,18 @@
 #'     arrow = arrow(length = unit(0.1,"cm"))) +
 #'   borders("state")
 #'
+#' # Use lineend and linejoin to change the style of the segments
+#' df2 <- expand.grid(lineend = c('round', 'butt', 'square'),
+#'                    linejoin = c('round', 'mitre', 'bevel'), stringsAsFactors = FALSE)
+#' df2 <- data.frame(df2, y = 1:9)
+#' ggplot() +
+#'   geom_segment(aes(x = 1, y = y, xend = 2, yend = y), df2, size = 4,
+#'                lineend = df2$lineend, linejoin = df2$linejoin,
+#'                arrow = arrow(length = unit(0.5, "inches"))) +
+#'   geom_text(aes(x = 2, y = 1:9, label = paste(df2$lineend, df2$linejoin)),
```

Comment:
I think you can simplify this plot by setting `data` and `aes(y = y)` in the ggplot2 call.  The `geom_text()` layer shouldn't use `df$`